includes/class-wc-subscription.php

```diff
@@ -2107,27 +2113,45 @@ public function get_related_orders( $return_fields = 'ids', $order_types = array
 	/**
 	 * Get the related order IDs for a subscription based on an order type.
 	 *
-	 * @param string $order_type Can include 'any', 'parent', 'renewal', 'resubscribe' and/or 'switch'. Defaults to 'any'.
-	 * @return array List of related order IDs.
 	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v2.3.0
+	 * @since 7.2.1 - The $order_type parameter can now be an array of order types and the $return_type parameter was added.
+	 *
+	 * @param string|array $order_type  Can include 'any', 'parent', 'renewal', 'resubscribe' and/or 'switch'. Defaults to 'any'.
+	 * @param string       $return_type The format to return the related order IDs in. Can be 'flat' or 'grouped'. Defaults to 'flat'.
+	 *
+	 * @return array List of related order IDs.
 	 */
-	protected function get_related_order_ids( $order_type = 'any' ) {
-
-		$related_order_ids = array();
+	protected function get_related_order_ids( $order_type = 'any', $return_type = 'flat' ) {
+		$order_types       = is_array( $order_type ) ? $order_type : [ $order_type ];
+		$related_order_ids = [];
 
-		if ( in_array( $order_type, array( 'any', 'parent' ) ) && $this->get_parent_id() ) {
-			$related_order_ids[ $this->get_parent_id() ] = $this->get_parent_id();
+		// For backwards compatibility, replace 'any' with the actual order types.
+		if ( in_array( 'any', $order_types, true ) ) {
+			$order_types = array_diff( $order_types, [ 'any' ] ); // Remove 'any'.
+			$order_types = array_unique( array_merge( $order_types, [ 'parent', 'renewal', 'resubscribe', 'switch' ] ) ); // Add the 'any' order types.
 		}
 
-		if ( 'parent' !== $order_type ) {
+		// Get the parent order ID first.
+		if ( in_array( 'parent', $order_types, true ) ) {
+			// Remove the parent order type from the list of order types.
+			$order_types = array_diff( $order_types, [ 'parent' ] );
+			$parent_id   = $this->get_parent_id();
 
-			$relation_types = ( 'any' === $order_type ) ? array( 'renewal', 'resubscribe', 'switch' ) : array( $order_type );
-
-			foreach ( $relation_types as $relation_type ) {
-				$related_order_ids = array_merge( $related_order_ids, WCS_Related_Order_Store::instance()->get_related_order_ids( $this, $relation_type ) );
+			if ( $parent_id ) {
+				$related_order_ids['parent'] = [ $parent_id ];
```

Comment:
This is another minor thing I noticed but this function doesn't return an empty 'parent' array if there's no parent order, but it does for empty 'renewal', 'switch' etc.

For example, if you create an New Subscription via WP admin with no parent or renewals and call:

```
wcs_get_subscription( 123 )->get_related_orders( 'all', [ 'parent', 'renewal', 'switch' ] );
```

The `get_related_order_ids()` returns an array like:

```
Array
(
    [renewal] => Array
        (
        )

    [switch] => Array
        (
        )

)
```

Not sure if we should add an empty array for parent if the calling function is requesting it?

Reply:
Oh good catch. I think this actually caused a slight breakage in backwards compatibility. 

Prior to this PR if you hooked onto the `woocommerce_subscription_related_orders` filter you'd get the opportunity to filter the parent order result. 

Because I don't return the empty array for `parent` it actually never triggers that filter anymore. I'll get that fixed up.

Reply:
Fixed in [ca71c96](https://github.com/Automattic/woocommerce-subscriptions-core/pull/790/commits/ca71c96b845253642cc6539552644f9e8a1bbc92)